components/script/links.rs

```diff
@@ -440,6 +440,10 @@ pub(crate) fn follow_hyperlink(
             referrer_policy,
             Some(secure),
             Some(document.insecure_requests_policy()),
+            document.has_trustworthy_ancestor_origin(),
+            // sebsebmc: investigate the ancestor, for a top level load we
```

Comment:
This is a downside of having the `has_trustworthy_ancestor_origin()` value also include the document's URL's origin's trustworthiness. If Document has both `has_trustworthy_ancestor_or_current_origin` and `has_trustworthy_ancestor_origin`, most callers could use the first one and code like this could use the second.

Reply:
I've gone through and changed things to do as you suggested.